docs/asf-docs/Introduction_hyp3_mintpy.ipynb

```diff
@@ -0,0 +1,527 @@
+    "2.3.2 clip the geotiff files in the current directory and its subdirectories"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define cutGeotiffsByLine function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from __future__ import print_function, absolute_import, division, unicode_literals\n",
+    "from osgeo import gdal\n",
+    "\n",
+    "def open_gdal_file(filename):\n",
+    "    handle = gdal.Open(filename)\n",
+    "    return handle\n",
+    "\n",
+    "def read_gdal_file_geo(filehandle, band=1):\n",
+    "    geotransform = filehandle.GetGeoTransform()\n",
+    "    geoproj = filehandle.GetProjection()\n",
+    "    return filehandle.RasterXSize, filehandle.RasterYSize, geotransform, geoproj\n",
+    "\n",
+    "def getOrigins(files):\n",
+    "\n",
+    "    ul = np.zeros((2,len(files)))\n",
+    "    lr = np.zeros((2,len(files)))\n",
+    "\n",
+    "    for i in range(len(files)):\n",
+    "        x,y,trans,proj = read_gdal_file_geo(open_gdal_file(files[i]))\n",
+    "        ul[0,i] = trans[0]\n",
+    "        lr[0,i] = trans[0] + x*trans[1]\n",
+    "        ul[1,i] = trans[3]\n",
+    "        lr[1,i] = trans[3] + y*trans[5]\n",
+    "\n",
+    "    return ul,lr,trans[1],trans[5]\n",
+    "\n",
+    "def copyOrigins(files,all_coords,all_pixsize):\n",
+    "    ul = np.zeros((2,len(files)))\n",
+    "    lr = np.zeros((2,len(files)))\n",
+    "\n",
+    "    for i in range(len(files)):\n",
+    "        coords = all_coords[i]\n",
+    "        ul[0,i] = coords[0]\n",
+    "        lr[0,i] = coords[2]\n",
+    "        ul[1,i] = coords[1]\n",
+    "        lr[1,i] = coords[3]\n",
+    " \n",
+    "        if i == 0:\n",
+    "            xres = all_pixsize[i]\n",
+    "            yres = all_pixsize[i]\n",
+    "\n",
+    "    return ul,lr,xres,yres\n",
+    "\n",
+    "def cutGeotiffsByLine(files):\n",
+    "    ul,lr,xres,yres = getOrigins(files)\n",
+    "    \n",
+    "    diff_ul = np.zeros((2,len(files)))\n",
+    "    diff_ul[0] = (max(ul[0])-ul[0])/xres    \n",
+    "    diff_ul[1] = -1*(min(ul[1])-ul[1])/(-1*yres)\n",
+    "\n",
+    "    print(\"Difference list:\")\n",
+    "    print(diff_ul)\n",
+    "\n",
+    "    lrx = min(lr[0])\n",
+    "    lry = max(lr[1])\n",
+    "    lenx = (lrx-max(ul[0])) / xres\n",
+    "    leny = -1*(lry-min(ul[1])) / (-1*yres)\n",
+    "    if leny < 0:\n",
+    "        leny = abs(leny)\n",
+    "        diff_ul[1] = diff_ul[1] * -1\n",
+    "    print(\"Size of output images {} x {}\".format(lenx, leny))\n",
+    "\n",
+    "    outfiles = []   \n",
+    "    for i in range(len(files)):\n",
+    "        outfile = files[i].replace(\".tif\",\"_clip.tif\")\n",
+    "        print(\"Processing file {} to create file {}\".format(files[i], outfile))\n",
+    "        gdal.Translate(outfile,files[i],srcWin=[diff_ul[0,i],diff_ul[1,i],lenx,leny],noData=0)\n",
```

Comment:
No reason to override the noData value here, better to leave it out and let the output tif have the same as the input tif (if any)
```suggestion
    "        gdal.Translate(outfile,files[i],srcWin=[diff_ul[0,i],diff_ul[1,i],lenx,leny])\n",
```

Reply:
changed